### DISCUSSION TITLE
Need line cap style for rounded corner of line like visio.

### DISCUSSION THREAD

**Discussion by gautamyadav144** [jgraph/drawio#4200] [General]
I want to use rounded corner line , is it possible?

**Comment by masa300V** [comment]
May I know the solution here as you closed as "completed"?  I'm looking for the way changing linecap (and linejoin) of line shapes.

**Reply by gigberg** [reply]
see as [https://webapps.stackexchange.com/questions/66736/how-to-draw-right-angle-connectors-and-endpoints-with-draw-io](https://webapps.stackexchange.com/questions/66736/how-to-draw-right-angle-connectors-and-endpoints-with-draw-io)

**Reply by masa300V** [reply]
@gigberg  thanks for the link...but...

I'm looking for a line-cap style of "round" so to connect 2 lines smoothly.
![image](https://github.com/jgraph/drawio/assets/37952014/5b96b020-da09-4425-8077-d5f5efd662e5)

Why I insist?  I make a electronics symbols (like resistors, coils, caps), placing those symbols (electronics components), and then, connect those symbols with another lines (to me wires)...so the "round" cap is the only solution for a smooth 90deg connection.

**Comment by masa300V** [comment]
Very recently, I found draw.io is the tool I keep looking for, almost near-perfect.
I use drawing tools (so far Visio) to generate my electronics circuit diagram for presentations.

***

When drawing electronics diagrams, Very often I need to connect line to line.  In this case, linecap matters and (in any drawing tool) the "round" linecap is the solution.

![image](https://github.com/jgraph/drawio/assets/37952014/35248b5b-e4ce-455d-8ae8-1508179c9854)

Zooming in my coil connection of 90deg line connection...  looks not a serious thing but this kind of no-smooth connection catching people's eyes to impress as "hard to review diagram".

![image](https://github.com/jgraph/drawio/assets/37952014/a3c78259-f4d5-4efe-9330-2b06d86311df)

***

Please consider to implement linecap (and hopefully linejoin).
If it makes sense, it's OK draw.io GUI not rendering linecap as long as property of linecap goes to the output svg or png file so that my final HTML or PDF viewer can render "round" shape :-P

**Comment by davidjgraph** [comment]
Line Join added in 24.2.7.

**Reply by masa300V** [reply]
Thanks a lot!  I'm waiting for the "desktop" version of 24.2.7 from Microsoft Store.
